main/src/library/FilePathWithError.flix

```diff
@@ -0,0 +1,481 @@
+                } catch {
+                    case ex: InvalidPathException => Err(IOError.Generic(ex.getMessage()))
+                    case ex: IOException => Err(IOError.Generic(ex.getMessage()))
+                };
+                k(res)
+            }
+
+            def creationTime(f, k) = {
+                let res = try {
+                    Ok(fileAttributes(Paths.get(f)).creationTime().toMillis())
+                } catch {
+                    case ex: InvalidPathException => Err(IOError.Generic(ex.getMessage()))
+                    case ex: IOException => Err(IOError.Generic(ex.getMessage()))
+                };
+                k(res)
+            }
+
+            def modificationTime(f, k) = {
+                let res = try {
+                    Ok(fileAttributes(Paths.get(f)).lastModifiedTime().toMillis())
+                } catch {
+                    case ex: InvalidPathException => Err(IOError.Generic(ex.getMessage()))
+                    case ex: IOException => Err(IOError.Generic(ex.getMessage()))
+                };
+                k(res)
+            }
+
+            def size(f, k) = {
+                let res = try {
+                    Ok(Files.size(Paths.get(f)))
+                } catch {
+                    case ex: InvalidPathException => Err(IOError.Generic(ex.getMessage()))
+                    case ex: IOException => Err(IOError.Generic(ex.getMessage()))
+                };
+                k(res)
+            }
+
+            def read(f, k) = {
+                let res = try {
+                    Ok(new JString(Files.readAllBytes(Paths.get(f))))
+                } catch {
+                    case ex: InvalidPathException => Err(IOError.Generic(ex.getMessage()))
+                    case ex: IOException => Err(IOError.Generic(ex.getMessage()))
+                };
+                k(res)
+            }
+
+            def readLines(f, k) = {
+                let res = try {
+                    Ok(FromJava.fromJava(Files.readAllLines(Paths.get(f))))
+                } catch {
+                    case ex: InvalidPathException => Err(IOError.Generic(ex.getMessage()))
+                    case ex: IOException => Err(IOError.Generic(ex.getMessage()))
+                };
+                k(res)
+            }
+
+            def readBytes(f, k) = {
+                let res = try {
+                    Ok(Array.toVector(Files.readAllBytes(Paths.get(f))))
+                } catch {
+                    case ex: InvalidPathException => Err(IOError.Generic(ex.getMessage()))
+                    case ex: IOException => Err(IOError.Generic(ex.getMessage()))
+                };
+                k(res)
+            }
+
+            def write(data, f, k) = {
+                let res = try {
+                    let bytes = data#str.getBytes(StandardCharsets.UTF_8);
+                    Files.write(Paths.get(f), bytes, (...{}: Vector[OpenOption]));
+                    k(())
```

Comment:
Why call k here? (I am also surprised it type checks).

Reply:
(This pattern repeats-- but k should only be called once, and it should always be called.).

Reply:
> Why call k here? (I am also surprised it type checks). (EDIT: Its because both args are Unit)

But the continuation takes `Result[IOError, Unit]`, right? So why can we just pass it unit?

Reply:
You figured it out right? :)

Reply:
I made it work, but why was this not caught?

Reply:
I think there are still some bugs (?)